src/main/java/hudson/plugins/gradle/injection/MavenBuildScanInjection.java

```diff
@@ -82,12 +83,9 @@ private void injectMavenExtensions(Node node, FilePath nodeRootPath) {
             LOGGER.info("Injecting Maven extensions " + nodeRootPath);
             List<FilePath> libs = new LinkedList<>();
 
-            extensionsHandler.copyGradleEnterpriseExtensionToAgent(nodeRootPath);
-            libs.add(extensionsHandler.getGradleEnterpriseExtensionPath(nodeRootPath));
-
+            libs.add(extensionsHandler.copyGradleEnterpriseExtensionToAgent(nodeRootPath));
             if (getGlobalEnvVar(GE_CCUD_VERSION_VAR) != null) {
```

Comment:
Since we are no longer doing a cleanup before I think this should be cleaned up if CCUD injection was disabled.

Reply:
Add deletion logic.